pypeit/core/slitdesign_matching.py

```diff
@@ -0,0 +1,229 @@
+    sigres = sigma_clipped_stats(res, sigma = sigrej)[2]   # RMS residuals
+    cut = 5 if res.size<5 else sigrej*sigres
+    out = numpy.abs(res)> cut
+
+    # check for duplicate indices
+    bad = numpy.ones(ind.size, dtype=int)
+    bad[numpy.unique(ind, return_index=True)[1]] = 0
+    wbad = numpy.where(bad)[0]
+    if wbad.size > 0:
+        print('Trimming {} bad matches in deimos_slit_match'.format(wbad.size))
+        for i in range(wbad.size):
+            badind = ind[wbad[i]]
+            w = ind == badind
+            bad[w] = 1    # set all with (ind eq badind) bad for a moment
+            wdif = numpy.argmin(numpy.abs(res[w]))
+            bad[w[wdif]] = 0  # set this one not bad
+        bad = bad|out
+        good = bad == 0
+        ind = ind[good]
+        x_det=x_det[good]
+    if print_matches is True:
+        if edge is not None:
+            print('-----------------------------------------------')
+            print('             {} slit edge               '.format(edge))
+        print('-----------------------------------------------')
+        print('Index      xblupix       found_x               ')
+        print('-----------------------------------------------')
+        for i in range(ind.size):
+            print('{}  {}  {}'.format(ind[i], x_model[ind][i], x_det[i]))
+    return x_det, ind, coeff, sigres
+
+
+
+def plot_matches(edgetrace, ind, x_model, x_det, yref, slit_index, edge=None):
+    yref_xdet=numpy.tile(yref, x_det.size)
+    yref_x_model=numpy.tile(yref, x_model.size)
+    plt.rc('xtick', direction='in')
+    plt.rc('ytick', direction='in')
+    fig=plt.figure(figsize=(10, 4.5))
+    fig.subplots_adjust(left=None, bottom=None, right=None, top=None, wspace=0, hspace=0)
+    if edge is not None:
+        plt.title('{} slit edges cross-matching'.format(edge))
+    for x in edgetrace.T:
+        plt.plot(x, numpy.arange(4096), color='k', lw=0.5, zorder=0)
+    plt.scatter(x_det, yref_xdet, marker='D', s=10, lw=0, color='m', zorder=1,
+                                      label='Image trace midpoint (x_det)')
+    plt.scatter(x_model, yref_x_model, marker='o', s=10, lw=0, color='b', zorder=1,
+                                      label='Optical model trace BEFORE x-corr (x_model)')
+    plt.scatter(x_model[ind], yref_x_model[ind], marker='o', s=40, facecolors='none', edgecolors='g', zorder=1,
+                                      label='Optical model trace AFTER x-corr (x_model[ind])')
+    for i in range(x_model[ind].size):
+        plt.text(x_model[ind][i], yref_x_model[ind][i]+100, slit_index[ind][i], rotation=45, color='g',
+                                     fontsize=8, horizontalalignment='center')
+    for i in range(x_model.size):
+        plt.text(x_model[i], yref_x_model[i]-300, slit_index[i], rotation=45, color='b',
+                                     fontsize=8, horizontalalignment='center')
+    plt.xlabel('Spatial pixels')
+    plt.ylabel('Spectral pixels')
+    plt.xlim(0, 2218)
```

Comment:
These appear to be hard coded to DEIMOS. Grab the sizes of the arrays you are using in this code and decide based on them how to scale the axes.

Reply:
Done.